Mathlib/NumberTheory/Padics/PadicVal.lean

```diff
@@ -589,6 +593,29 @@ theorem padicValNat_factorial {p n b : ℕ} [hp : Fact p.Prime] (hnb : log p n <
   PartENat.natCast_inj.mp ((padicValNat_def' (Nat.Prime.ne_one hp.out) <| factorial_pos _) ▸
       Prime.multiplicity_factorial hp.out hnb)
 
+/-- **Legendre's Theorem**
+
+Taking (`p - 1`) times the `p`-adic valuation of `n!` equals `n` minus the sum of base `p` digits
+of `n`. -/
+theorem padicValNat_factorial' {p : ℕ} [hp : Fact p.Prime] (n : ℕ):
+    (p - 1) * padicValNat p n ! = n - (p.digits n).sum := by
+  apply Nat.strongInductionOn n
+  intro n hn
+  by_cases n = 0
```

Comment:
```suggestion
  rcases n.eq_zero_or_pos with rfl | hnp; simp
```

Reply:
Note that according to the [style guide](https://leanprover-community.github.io/contribute/style.html#tactic-mode)

> Very short goals can be closed right away using swap or work_on_goal if needed, to avoid additional indentation in the rest of the proof.

so it is acceptable, and I claim preferable, to dispense with the `n = 0` branch using `; simp`

You should then drop the `· ` and remove the intendation from the rest of the proof.

Reply:
done